src/server/twitterSearch.js

```diff
@@ -15,51 +16,161 @@ var T = new Twit({
+    return linkTweeterToRetweet(db, retweeter, originalTweet);
+  });
+}
 
-      upsertTweeter(db, tweeter).then((result) => {
-        upsertTweet(db, tweet).then((result) => {
-          linkTweeterToTweet(db, tweeter, tweet);
-        });
+/**
+ * Link a tweet to all of its hashtags
+ * @param db The OrientDB instance
+ * @param rawHashtags The original hashtags
+ * @param tweet
+ * @returns {Promise}
+ */
+const linkTweetToHashtags = (db, rawHashtags, tweet) => {
+  return Promise.all(
+    rawHashtags.map((rawHashtag) => {
+      const hashtag = Builders.HashtagBuilder().content(rawHashtag.text.toLowerCase()).build();
+      return upsertHashtag(db, hashtag).then((result) => {
+        return linkTweetToHashtag(db, tweet, hashtag);
       });
+    })
+  );
+};
 
-      tweetRaw.entities.hashtags.forEach((hashtagRaw) => {
-        const hashtag = Builders.HashtagBuilder().content(hashtagRaw.text.toLowerCase()).build();
-        upsertHashtag(db, hashtag).then((result) => {
-          linkTweetToHashtag(db, tweet, hashtag);
-        });
+const linkTweetToMentions = (db, rawMentions, tweet) => {
+  return Promise.all(
+    rawMentions.map((rawMention) => {
+      const mentionedTweeter = buildTweeterFromRaw(rawMention);
+      return upsertTweeter(db, mentionedTweeter).then((result) => {
+        return linkTweetToTweeterViaMention(db, tweet, mentionedTweeter);
       });
+    })
+  );
+};
+/**
+ * Given a raw status we know is not a retweet, insert it and upsert the user.
+ * @param db The OrientDB instance
+ * @param rawTweet A raw status from the Twitter API
+ * @param originalTweeter An immutable Tweeter object
+ * @returns {Promise}
+ */
+function processRawOriginalTweet(db, rawTweet, originalTweeter) {
+  const tweet = buildTweetFromRaw(rawTweet);
+  const rawHashtags = rawTweet.entities.hashtags;
+  const rawMentions = rawTweet.entities.hashtags;
 
-      tweetRaw.entities.user_mentions.forEach((mentionRaw) => {
-        const mentionedTweeter = makeTweeterFromRaw(mentionRaw);
-        upsertTweeter(db, mentionedTweeter).then((result) => {
-          linkTweetToTweeterViaMention(db, tweet, mentionedTweeter);
-        });
-      });
+  return chainPromises(() => {
+    return upsertTweet(db, tweet);
+  }).then(() => {
+    return upsertTweeter(db, originalTweeter);
+  }).then(() => {
+    return linkTweeterToTweet(db, originalTweeter, tweet);
+  }).then(() => {
+    return linkTweetToHashtags(db, rawHashtags, tweet);
+  }).then(() => {
+    return linkTweetToMentions(db, rawMentions, tweet);
+  });
+}
+
+/**
+ * Given a raw tweet, extract information about the tweeter,
+ * if it was a retweet, etc, and store all information in
+ * the database.
+ * @param db The OrientDB instance
+ * @param rawTweet The original status object from the Twitter API
+ * @returns {Promise.<T>}
+ */
+const processTweet = (db, rawTweet) => {
+  const tweeter = buildTweeterFromRaw(rawTweet.user);
+  const rawRetweetedStatus = rawTweet.retweeted_status;

```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'retweeted_status' is not in camel case.](https://www.codacy.com/app/winwardo/solid-octo-disco/file/2175948194/issues/source?bid=3148782&fileBranchId=3163167#l130)